crates/engine/src/exec_context.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::types::{
+    AccountId,
+    Balance,
+};
+
+/// The context of a contract execution.
+#[cfg_attr(test, derive(Debug, PartialEq, Eq))]
+pub struct ExecContext {
+    /// The caller of the contract execution.
+    ///
+    /// Might be user or another contract.
+    pub caller: Option<AccountId>,
+    /// The callee of the contract execution.
+    pub callee: Option<AccountId>,
```

Comment:
Since we don't know the specifics of the `AccountId` ‒ like how many bytes or what type of default `AccountId` makes sense they are left to be initialized by the crate which uses the `engine`. Methods which require a caller/callee panic when it has not been set (e.g. `fn caller()`). I'll add a comment explaning this.